Microsoft.Toolkit.Win32/Microsoft.Toolkit.Win32.UI.XamlHost/MetadataProviderDiscovery.cs

```diff
@@ -0,0 +1,84 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Reflection;
+
+namespace Microsoft.Toolkit.Win32.UI.XamlHost
+{
+    /// <summary>
+    /// MetadataProviderDiscovery is responsible for loading all metadata providers for custom UWP XAML
+    /// types.  In this implementation, reflection is used at runtime to probe for metadata providers in
+    /// the working directory, allowing any type that includes metadata (compiled in to a .NET framework
+    /// assembly) to be used without explicit metadata handling by the application developer.  This
+    /// internal class will be amended or removed when additional type loading support is available.
+    /// </summary>
+    internal static class MetadataProviderDiscovery
+    {
+        /// <summary>
+        /// Probes working directory for all available metadata providers
+        /// </summary>
+        /// <param name="filteredTypes">Types to ignore</param>
+        /// <returns>List of UWP XAML metadata providers</returns>
+        internal static List<Windows.UI.Xaml.Markup.IXamlMetadataProvider> DiscoverMetadataProviders(List<Type> filteredTypes)
+        {
+            // List of discovered UWP XAML metadata providers
+            List<Windows.UI.Xaml.Markup.IXamlMetadataProvider> metadataProviders = null;
+            metadataProviders = new List<Windows.UI.Xaml.Markup.IXamlMetadataProvider>();
+
+            // Reflection-based runtime metadata probing
+            var currentDirectory = System.IO.Directory.GetCurrentDirectory();
```

Comment:
@ryalanms Working directory is different when packaged inside a Centennial App. This takes forever to load when a WPF app that uses Islands is packaged inside an APPX. We should handle APPX scenarios.

Reply:
@azchohfi I've worked out a fix for the issue in 9e3ea34